harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -778,16 +785,20 @@ def _train_for_op(
                     self._upload_tb_files()
                 self._stop_requested()
 
+                # Test mode will break after one batch despite not completing op.
+                if self.is_chief and not self.test_mode:
+                    if not op._completed:
+                        # The only case where op isn't reported as completed is if we restarted but
+                        # op.length was already trained for and validated on; in that case just break
+                        # out of the operations loop; we have nothing to do.
+                        break
+
         # Finished training for op. Perform final checkpoint/validation if necessary.
         if not self._validation_is_current():
             self._validate(op)
         if not self._checkpoint_is_current():
             self._checkpoint(already_exiting=False)
 
-        # Test mode will break after one batch despite not completing op.
```

Comment:
why was this check moved up?

Reply:
It was a mistake moving it up, 

moved it back down but then removed the check since it doesn't do anything but cause errors? It's not in a loop to break out early.

Reply:
the check is on the `op` though, not the steps math in `self._steps_until_complete(searcher_length) > 0`. before, it would have errored in the case where `steps_until_complete(searcher_length) <= 0` but the `op` was not reported as complete, and prevented the parent `for op in searcher_ops` loop from continuing. 

so it's not a big deal but in this case i think we should still keep the check, but change the assert to `raise ShouldExit()` to essentially break out of the parent caller's `for op in ops` loop.

Reply:
Makes sense, changed the assert to raise ```ShouldExit```